engine/runtime-integration-tests/src/test/java/org/enso/interpreter/test/ContextInitTest.java

```diff
@@ -0,0 +1,41 @@
+package org.enso.interpreter.test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import org.enso.test.utils.ContextUtils;
+import org.enso.test.utils.ProjectUtils;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class ContextInitTest {
+  @ClassRule public static final TemporaryFolder tmpFolder = new TemporaryFolder();
+
+  @Test
+  public void shouldChangeWorkingDir_BeforeExecutingProject() throws IOException {
+    var projDir = tmpFolder.newFolder().toPath();
+    var mainSrc =
+        """
+        from Standard.Base import all
+        main =
+            file = File.new "MY_FILE.txt"
+            file.absolute.path
+        """;
+    ProjectUtils.createProject("Project", mainSrc, projDir);
+    var out = new ByteArrayOutputStream();
+    var ctxBuilder = ContextUtils.defaultContextBuilder().out(out).err(out);
+    var expectedWorkingDir = projDir.getParent();
+    var expectedPath = expectedWorkingDir.resolve("MY_FILE.txt").toAbsolutePath().toString();
+    ProjectUtils.testProjectRun(
```

Comment:
Better to turn this (also) into integration test that launches `enso` with and without `--jvm` flag. Such tests are traditionally placed into `Example_Tests` - like https://github.com/enso-org/enso/blob/ed63362051842cdff6391423eeb6f33d3371df87/test/Examples_Tests/src/Python_Examples_Spec.enso#L91

Reply:
As noted in the commit message of https://github.com/enso-org/enso/pull/12618/commits/ae2b06c18a711a50b350176e509df1721f22a898, I have already tried to test it that way, but the test was failing in NI, as `enso_bin` could not be found. If you could think about a way how to launch something like `enso --new New_Project` from within enso running in NI, I will reintroduce that test once again.

Reply:
> There is no way of getting path to the currently running binary

There is a way. I was working with @sdedic to get such functionality into NI few years ago...

> how to launch something like enso --new New_Project from within enso running in NI

..one can `getClass().getProtectioDomain().getCodeSource().getLocation()` to get _"some URL"_ - probably the URL should point to the **location of `enso` executable**. Invoking that from a `.enso` test may need further changes:
- either some builtin 
- or reflective config

Reply:
At the end, there already was a test in [File_Spec](https://github.com/enso-org/enso/blob/2af704dd383ee89e19baf13db1c0c365ff5d350a/test/Base_Tests/src/System/File_Spec.enso#L377) that I re-enabled, and now is only ignored in JVM mode.